### PR TITLE
feat/Node: RewardsParameters to protocol

### LINKED ISSUE #1291: Add RewardsParameters to ProtocolParameters - Nodejs

### PR DESCRIPTION
**Pull request by kwek20**
# Description of change
Adds rewards parameters to node

## Links to any relevant issues

fixes #1291

### CODE REVIEW

File: bindings/nodejs/lib/types/models/info/node-info-protocol.ts [hunk 1]
```diff
@@ -102,6 +102,52 @@ export interface ProtocolParameters {
...
+/**
+ * Rewards Parameters defines the parameters that are used to calculate Mana rewards.
+ */
+export interface RewardsParameters {
+    /**
+     * The number of validation blocks that should be issued by a selected validator
+     * per slot during its epoch duties.
+     */
+    validationBlocksPerSlot: number;
+
```
Comment (Thoralf-M) [inline bindings/nodejs/lib/types/models/info/node-info-protocol.ts]: Why the extra spaces here?

Comment (kwek20) [reply bindings/nodejs/lib/types/models/info/node-info-protocol.ts]: ah well thats how i would write any interface :P but yea shouldve been like the others

File: bindings/nodejs/lib/types/models/info/node-info-protocol.ts [hunk 2]
```diff
@@ -102,6 +102,46 @@ export interface ProtocolParameters {
...
+     */
+    profitMarginExponent: number;
+    /**
+     * The length in epochs of the bootstrapping phase.
+     */
+    bootstrappingDuration: number;
+    /**
+     * Mana Share Coefficient is the coefficient used for calculation of initial rewards.
+     */
+    manaShareCoefficient: number;
```
Comment (Thoralf-M) [inline bindings/nodejs/lib/types/models/info/node-info-protocol.ts]: number string or whatever it's called

File: bindings/nodejs/lib/types/models/info/node-info-protocol.ts [hunk 3]
```diff
@@ -102,6 +102,46 @@ export interface ProtocolParameters {
...
+    /**
+     * Decay Balancing Constant Exponent is the exponent used for calculation of the initial reward.
+     */
+    decayBalancingConstantExponent: number;
+    /**
+     * Decay Balancing Constant is an integer approximation calculated based on chosen DecayBalancingConstantExponent.
+     */
+    decayBalancingConstant: number;
```
Comment (Thoralf-M) [inline bindings/nodejs/lib/types/models/info/node-info-protocol.ts]: number string or whatever it's called

File: bindings/nodejs/lib/types/models/info/node-info-protocol.ts [hunk 4]
```diff
@@ -102,6 +102,46 @@ export interface ProtocolParameters {
...
+     */
+    validationBlocksPerSlot: number;
+    /**
+     * Profit Margin Exponent is used for shift operation for calculation of profit margin.
+     */
+    profitMarginExponent: number;
+    /**
+     * The length in epochs of the bootstrapping phase.
+     */
+    bootstrappingDuration: number;
```
Comment (Thoralf-M) [inline bindings/nodejs/lib/types/models/info/node-info-protocol.ts:128]:
```suggestion
    bootstrappingDuration: u64;
```

Comment (kwek20) [reply bindings/nodejs/lib/types/models/info/node-info-protocol.ts:128]: this is length in epochs, so will be u32 soon right? Thats why i left it

Comment (Thoralf-M) [reply bindings/nodejs/lib/types/models/info/node-info-protocol.ts:128]: Ah yeah probably will also be an u32 then

**Review by Thoralf-M — DISMISSED** [review]

File: bindings/nodejs/lib/types/models/info/node-info-protocol.ts [hunk 5]
```diff
@@ -102,6 +102,46 @@ export interface ProtocolParameters {
...
+    /**
+     * Mana Share Coefficient is the coefficient used for calculation of initial rewards.
+     */
+    manaShareCoefficient: u64;
+    /**
+     * Decay Balancing Constant Exponent is the exponent used for calculation of the initial reward.
+     */
+    decayBalancingConstantExponent: number;
+    /**
+     * Decay Balancing Constant is an integer approximation calculated based on chosen DecayBalancingConstantExponent.
```
Comment (DaughterOfMars) [inline bindings/nodejs/lib/types/models/info/node-info-protocol.ts]:
```suggestion
     * Decay Balancing Constant is an integer approximation calculated based on chosen Decay Balancing Constant Exponent.
```

**Review by DaughterOfMars — DISMISSED** [review]

**Review by Thoralf-M — DISMISSED** [review]

File: bindings/nodejs/lib/types/models/info/node-info-protocol.ts [hunk 6]
```diff
@@ -102,6 +102,46 @@ export interface ProtocolParameters {
...
+
+/**
+ * Rewards Parameters defines the parameters that are used to calculate Mana rewards.
+ */
+export interface RewardsParameters {
+    /**
+     * The number of validation blocks that should be issued by a selected validator
+     * per slot during its epoch duties.
+     */
+    validationBlocksPerSlot: number;
```
Comment (thibault-martinez) [inline bindings/nodejs/lib/types/models/info/node-info-protocol.ts:120]: This is `validator_blocks_per_slot` in Rust so one of them is wrong

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

File: bindings/nodejs/lib/types/models/info/node-info-protocol.ts (hunk 6 above)
Comment (kwek20) [reply bindings/nodejs/lib/types/models/info/node-info-protocol.ts:120]: rust is, shall i update?

Comment (thibault-martinez) [reply bindings/nodejs/lib/types/models/info/node-info-protocol.ts:120]: Yes pls

Comment (thibault-martinez) [reply bindings/nodejs/lib/types/models/info/node-info-protocol.ts:120]: Has Python been done already? If yes, double check?

Comment (kwek20) [reply bindings/nodejs/lib/types/models/info/node-info-protocol.ts:120]: https://github.com/iotaledger/tips-draft/pull/25#issuecomment-1755703335

Comment (kwek20) [reply bindings/nodejs/lib/types/models/info/node-info-protocol.ts:119]: Lets make it validator_blocks_per_slot and change when they comment. no point keeping stuff open

Comment (kwek20) [reply bindings/nodejs/lib/types/models/info/node-info-protocol.ts:120]: This is ok now.

**Review by thibault-martinez — APPROVED** [review]

**Review by DaughterOfMars — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Pull request merged**